attributes/attributes.go

```diff
@@ -99,3 +104,35 @@ func (a *Attributes) Equal(o *Attributes) bool {
 	}
 	return true
 }
+
+// String prints the attribute map. If any key or values throughout the map
+// implement fmt.Stringer, it calls that method and appends.
+func (a *Attributes) String() string {
+	var sb strings.Builder
+	sb.WriteString("{")
+	var firstKey, firstVal string
+	var firstDone bool
+	for k, v := range a.m {
+		if !firstDone {
+			firstDone = true
+			if str, ok := k.(interface{ String() string }); ok {
+				firstKey = str.String()
+			}
+			if str, ok := v.(interface{ String() string }); ok {
+				firstVal = str.String()
+			}
+			continue
+		}
+		var key, val string
+		if str, ok := k.(interface{ String() string }); ok {
+			key = str.String()
+		}
+		if str, ok := v.(interface{ String() string }); ok {
+			val = str.String()
+		}
+		sb.WriteString(fmt.Sprintf("%q: %q, ", key, val))
+	}
+	sb.WriteString(fmt.Sprintf("%q: %q", firstKey, firstVal))
+	sb.WriteString("}")
+	return sb.String()
+}
```

Comment:
How about this?

```suggestion
func (a *Attributes) String() string {
	var sb strings.Builder
	sb.WriteString("{")
	first := true
	for k, v := range a.m {
		var key, val string
		if str, ok := k.(interface{ String() string }); ok {
			key = str.String()
		}
		if str, ok := v.(interface{ String() string }); ok {
			val = str.String()
		}
		if !first {
			sb.WriteString(", ")
		}
		sb.WriteString(fmt.Sprintf("%q: %q", key, val))
		first = false
	}
	sb.WriteString("}")
	return sb.String()
}
```

Reply:
This doesn't work. Your code would be {k1: v1 k2: v2, k3: v3,}. The only way to do this is to persist something from iteration and write at end? You can't really do last one so I decided to do first one that comes up in iteration to handle the case of a single attribute pair.

Reply:
How about this: https://go.dev/play/p/7ArpLBzuSGB

Reply:
Switched. Nice solution :).